x-pack/plugins/enterprise_search/server/index.ts

```diff
@@ -19,6 +19,12 @@ export const configSchema = schema.object({
   enabled: schema.boolean({ defaultValue: true }),
   accessCheckTimeout: schema.number({ defaultValue: 5000 }),
   accessCheckTimeoutWarning: schema.number({ defaultValue: 300 }),
+  ssl: schema.object({
+    certificateAuthorities: schema.maybe(
+      schema.oneOf([schema.arrayOf(schema.string()), schema.string()])
+    ),
+    rejectUnauthorized: schema.boolean({ defaultValue: false }),
```

Comment:
Using `defaultValue: false` here essentially disables certificate validation. E.g., all certificates will be trusted, regardless of what is specified in `certificateAuthorities` above.

If you want to use `rejectUnauthorized`, it should be `defaultValue: true`. See my other comment though, it may be better to use `verificationMode` instead.

Reply:
> Using defaultValue: false here essentially disables certificate validation

DOH, you are right, I misinterpreted the name. Should absolutely be `true` by default!